### PR TITLE
Add automerge-repo-react-hooks tests

### PR DESCRIPTION
**Pull request by neftaly**
This PR adds tests for `useRepo`, using jsdom to mock a browser, via vitest.

### CODE REVIEW

**Comment by neftaly** [comment]
I was hoping to use a headless browser, so that we can test polyfills for the web navigation API, unfortunately [this part of vitest](https://vitest.dev/guide/browser.html) is still experimental, and will not work until browser test isolation lands.

**vitest.config.ts:**
```
test: {
    browser: {
      enabled: true,
      headless: true,
      provider: 'playwright',
      name: 'firefox', // Navigation API only supported in Chrome; check FF polyfill works
    }
}
```

File: packages/automerge-repo-react-hooks/src/useRepo.test.tsx [hunk 1]
```diff
@@ -1,22 +1,33 @@
...
+      /Repo was not found on RepoContext/
+    )
 
-  //   expect(result).toBeInstanceOf(Repo)
+    spy.mockRestore()
   })
 
+  test.skip("should return repo from context", () => {
+    const wrapper = ({ children }) => (
+      <RepoContext value={repo} children={children} />
```
Comment (neftaly) [inline packages/automerge-repo-react-hooks/src/useRepo.test.tsx:28]: This does not work yet, once it does the PR will be ready for review

File: packages/automerge-repo/package.json [hunk 2]
```diff
@@ -34,7 +34,7 @@
   "dependencies": {
     "cbor-x": "^1.3.0",
     "debug": "^4.3.4",
-    "eventemitter3": "^4.0.7",
+    "eventemitter3": "^5.0.1",
```
Comment (neftaly) [inline packages/automerge-repo/package.json:37]: <https://github.com/primus/eventemitter3/releases/tag/5.0.0>

**Comment by neftaly** [comment]
Turns out I needed `<RepoContext.Provider>` not `<RepoContext>` -.-

**Comment by pvh** [comment]
Hey @neftaly, apologies for the delay in getting into this. I think for now we can disable this test in the test runner on the github action server but we'll need to figure out the best way to do that before I can land because it will either crash the tests or leave them hanging if we don't.

**Comment by neftaly** [comment]
Are you ok with renaming "yarn test" command to "yarn hooks-test"? Or do you want me to add config to actions?

**Comment by pvh** [comment]
ah, that's extremely simple and totally fine by me for a stop-gap

**Comment by pvh** [comment]
@neftaly I'm going to just do that for you since I don't see a patch yet and it's trivial, then merge. Thank you for this!

**Comment by pvh** [comment]
Done!

File: packages/automerge-repo-react-hooks/package.json [hunk 3]
```diff
@@ -10,11 +10,22 @@
   "main": "dist/index.js",
   "scripts": {
     "build": "tsc",
+    "test-browser": "vitest",
```
Comment (neftaly) [inline packages/automerge-repo-react-hooks/package.json:13]: this should be `"test": "vitest run",`

**Comment by neftaly** [comment]
Thanks!
I didn't know the CLI would pick up this test, sorry. The tests were running in watcher mode. Changing test command to `vitest run` will fix this.

**Pull request merged**